.lintstagedrc

```diff
@@ -0,0 +1,9 @@
+{
+  "*.{ts,js}": [
+    "node packages/build/bin/run-prettier --write",
```

Comment:
Can we only fix changed files for `prettier`?

Reply:
Can we also add a print out to describe how to skip the check if the command fails?

Reply:
`lint-staged` runs command only on changed files that were staged for commit by definition.

This is the output when pre-commit hook fails:

<details>
  <summary>output</summary>
  
```
~ git commit -a -m "chore: fail prettier"
husky > pre-commit (node v12.16.0)
[STARTED] Preparing...
[SUCCESS] Preparing...
[STARTED] Running tasks...
[STARTED] Running tasks for *.{ts,js}
[STARTED] Running tasks for *.{md}
[STARTED] node packages/build/bin/run-prettier --write
[FAILED] node packages/build/bin/run-prettier --write [FAILED]
[FAILED] node packages/build/bin/run-prettier --write [FAILED]
[SUCCESS] Running tasks...
[STARTED] Applying modifications...
[STARTED] Reverting to original state because of errors...
[SUCCESS] Reverting to original state because of errors...
[STARTED] Cleaning up...
[SUCCESS] Cleaning up...

× node packages/build/bin/run-prettier --write:
[error] packages\repository\src\repositories\legacy-juggler-bridge.ts: SyntaxError: ';' expected. (464:8)
[error]   462 |   }
[error]   463 |
[error] > 464 |   fail here
[error]       |        ^
[error]   465 |
[error]   466 |   async updateById(
[error]   467 |     id: ID,
husky > pre-commit hook failed (add --no-verify to bypass)
``` 
</details>

Is this sufficient?

Reply:
@InvictusMB I understand the command is only triggered for changes files. But `node packages/build/bin/run-prettier --write` itself scans all files.

Reply:
I'm referring the possibility of https://github.com/strongloop/loopback-next/blob/63e8f6e1f91b0808696a11b123318d5089f2b2d0/lint-staged.config.js#L13.

Reply:
The output is good enough to understand what's failing. My comment was to print out an option to skip the failure so that the pre-commit hooks can be skipped.

Reply:
> I'm referring the possibility of
> 
> https://github.com/strongloop/loopback-next/blob/63e8f6e1f91b0808696a11b123318d5089f2b2d0/lint-staged.config.js#L13

Lint staged literally does that for us already.
That's what gets passed into `run-prettier`
```
run-prettier argv
[
  'C:\\DevTools\\nodejs\\node.exe',
  'D:\\Projects\\loopback-next\\packages\\build\\bin\\run-prettier',
  '--write',
  'D:/Projects/loopback-next/packages/build/bin/run-prettier.js',
  'D:/Projects/loopback-next/packages/repository/src/repositories/legacy-juggler-bridge.ts'
]
```

> The output is good enough to understand what's failing. My comment was to print out an option to skip the failure so that the pre-commit hooks can be skipped.

It is the last line in the output:
```
husky > pre-commit hook failed (add --no-verify to bypass)
```
So far there does not seem to be a way to customize that message.
If you mean highliting `LINT_STAGED=false` option, I'm not sure we should also promote a permanent switch here. It is described in the docs and visible in pre-commit hook definition in `package.json`.